distributed/shuffle/_worker_extension.py

```diff
@@ -524,30 +752,48 @@ async def _refresh_shuffle(
     async def _refresh_shuffle(
         self,
         shuffle_id: ShuffleId,
-        empty: pd.DataFrame,
-        column: str,
-        npartitions: int,
+        type: str,
+        kwargs: dict,
     ) -> ShuffleRun:
         ...
 
     async def _refresh_shuffle(
         self,
         shuffle_id: ShuffleId,
-        empty: pd.DataFrame | None = None,
-        column: str | None = None,
-        npartitions: int | None = None,
+        type: str | None = None,
+        kwargs: dict | None = None,
     ) -> ShuffleRun:
         import pyarrow as pa
 
-        result = await self.worker.scheduler.shuffle_get(
-            id=shuffle_id,
-            schema=pa.Schema.from_pandas(empty).serialize().to_pybytes()
-            if empty is not None
-            else None,
-            npartitions=npartitions,
-            column=column,
-            worker=self.worker.address,
-        )
+        if type is None:
+            result = await self.worker.scheduler.shuffle_get(
+                id=shuffle_id,
+                worker=self.worker.address,
+            )
+        elif type == "DataFrameShuffle":
+            assert kwargs is not None
+            result = await self.worker.scheduler.shuffle_get_or_create(
+                id=shuffle_id,
+                type=type,
+                spec={
+                    "schema": pa.Schema.from_pandas(kwargs["empty"])
+                    .serialize()
+                    .to_pybytes(),
+                    "npartitions": kwargs["npartitions"],
+                    "column": kwargs["column"],
+                },
+                worker=self.worker.address,
```

Comment:
Why not just forward kwargs to the scheduler extension as is? On scheduler side we already have a similar switch statement and we could avoid some complexity. Is this because the arrow schema needs special serialization? Feels also a bit leaky but not super important

Reply:
Yeah, there's some custom serialization work that needs to happen on the worker (or does it really?) that prevents this from simply sending the kwargs over. I think I may want to move the generation of this `spec` into the actual `rechunk_transfer` task and introduce a `dataclass` that can handles transformation to a JSON-serializable message for us.

That's on my mental TODO list for follow-up PRs though.